csrc/evaluator_common.cpp

```diff
@@ -342,6 +343,23 @@ void PrecomputedValues::bindTensorMetaData(
       bindValue(extent->evaluatorIndex(), value);
     }
   }
+
+  // Here we bind TensorMetaData so that GetMetaData expressions can be
+  // evaluated. Note that we do not bind the at::Tensor itself here since that
+  // would mean PrecomputedValues will own the tensor. Unlike
+  // ExpressionEvaluator, PrecomputedValues objects are typically long-lived, so
+  // we do not want them to own large objects.
+  std::shared_ptr<Struct> struct_ = std::make_shared<TensorMetaData>();
+  TensorMetaData* metadata = (TensorMetaData*)struct_.get();
+  metadata->dtype =
+      std::get<PrimDataType>(aten_to_data_type(tensor.scalar_type()).type);
+  metadata->data = tensor.data_ptr();
+  metadata->logical_size = tensor.sizes();
+  metadata->logical_stride = tensor.strides();
+  metadata->alloc_size = tensor.sizes();
+  metadata->alloc_stride = tensor.strides();
```

Comment:
How should we handle these? Currently this is failing in the test `AllocationDomainTest.NHWC4d_To_NHWC4d_CUDA`. In that test we have a memcpy fusion with channels-last inputs and outputs and we schedule vectorization of size 4. I get the error
> terminate called after throwing an instance of 'nvfuser::nvfError'                                                               
  what():  is_contiguous || size == 1 || is_expanded_broadcasting || (still_rightmost && stride == 1) || (!still_rightmost && stride % word_size == 0) INTERNAL ASSERT FAILED at "/opt/pytorch/nvfuser/csrc/executor_utils.cpp":545, please report a bug with repro script to NVFuser at https://github.com/NVIDIA/Fuser/issues. Vectorization of T0_g[ iS15{( (( (( getMetaData(T0) )).logical_size ))[0] )}, iS16{( (( (( getMetaData(T0) )).logical_size ))[1] )}, iS17{( (( (( getMetaData(T0) )).logical_size ))[2] )}, iS18{( 
(( (( getMetaData(T0) )).logical_size ))[3] )} ] with word size 4 not possible due to invalid stride. Domain: iS18{( (( (( getMetaData(T0) )).logical_size ))[3] )}, stride: 21, cur_contig_stride: 1, non contig due to slice: 0

Inspecting this a bit I see that `logical_stride` and `alloc_stride` are the same because of the line above, but they should differ for this channels-last example.

Reply:
Sorry, missed this message. Is this still the case?

Reply:
Oh this comment is outdated now since I realize we should just use `ExpressionEvaluator` to get the metadata object here instead of trying (and failing) to replicate it as I did previously.